metafields.go

```diff
@@ -0,0 +1,196 @@
+package shopify
+
+import (
+	"strconv"
+	"time"
+)
+
+// Metafields are a flexible way to attach additional information to a Shopify resource (e.g. Product, Collection, etc.).
+type Metafields []Metafield
+
+// GetByKey gets a metafield via its key
+func (m Metafields) GetByKey(key string) (interface{}, error) {
+	var err error
+	for _, metafield := range m {
+		if metafield.Key == key {
+			switch metafield.Type {
+			case NumberIntegerMetaFieldType:
+				var value int64
+				if metafield.Value != "" {
+					value, err = strconv.ParseInt(metafield.Value, 0, 64)
+					if err != nil {
+						return 0, err
+					}
+				}
+
+				return value, nil
+			case BooleanMetaFieldType:
+				var value bool
+				value, err = strconv.ParseBool(metafield.Value)
+				if err != nil {
+					return false, err
+				}
+
+				return value, nil
+			default:
+				return metafield.Value, nil
```

Comment:
Is the default a `string` type? Might be better to have all types in their own `case` and leave the default for cases that you have not supported/ not supported at all?

Reply:
Will test this.

Reply:
So I have opted for the default string because otherwise, this would be a massive switch statement and everything is a string other than two types. It just makes sense to assume a default as a string.